Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -254,82 +262,63 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
 
 // MARK: - Functions
 
-private func rewardsData(
-  _ data: [RewardAddOnCardViewData],
-  updatingQuantities quantities: SelectedRewardQuantities
-) -> [RewardAddOnSelectionDataSourceItem] {
-  guard !data.isEmpty else {
-    return [.emptyState(.addOnsUnavailable)]
-  }
-
-  return data.map { datum in
-    // Drill down into the reward and update its selectedQuantity
-    let quantity = quantities[datum.reward.id] ?? datum.reward.addOnData?.selectedQuantity
-
-    return RewardAddOnCardViewData(
-      project: datum.project,
-      reward: datum.reward
-        |> Reward.lens.addOnData .~ (
-          datum.reward.addOnData ?|> \.selectedQuantity .~ (quantity ?? 0)
-        ),
-      context: datum.context,
-      shippingRule: datum.shippingRule
-    )
-  }
-  .map(RewardAddOnSelectionDataSourceItem.rewardAddOn)
+private func unpack(
+  rewardsData: ([Reward], Project, Reward, PledgeViewContext, ShippingRule?),
+  selectedQuantities: SelectedRewardQuantities
+) -> ([Reward], Project, Reward, PledgeViewContext, ShippingRule?, SelectedRewardQuantities) {
+  return (rewardsData.0, rewardsData.1, rewardsData.2, rewardsData.3, rewardsData.4, selectedQuantities)
```

Comment:
Should we consider a model if there are 4 params in this tuple?

Reply:
In this case this function `unpack` just serves the purpose of taking a bunch of combines signals and returning it in the structure that we need it, so the call-site is neater:

```swift
self.loadAddOnRewardsIntoDataSource = rewardAddOnCardsViewData
  .takePairWhen(latestSelectedQuantities)
  .map(unpack) // this function performs the transformation
  .map(rewardsData) // this one returns `[RewardAddOnSelectionDataSourceItem]` for the data source
```